clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -439,6 +439,109 @@ pos_c"
     )
 }
 
+#[test]
+fn suggest_argument_multi_values() {
+    let mut cmd = Command::new("dynamic")
+        .arg(
+            clap::Arg::new("certain_num")
+                .long("certain_num")
+                .short('Y')
+                .value_parser(["val1", "val2", "val3"])
+                .num_args(3),
+        )
+        .arg(
+            clap::Arg::new("uncertain_num")
+                .long("uncertain_num")
+                .short('N')
+                .value_parser(["val1", "val2", "val3"])
+                .num_args(1..=3),
+        );
+
+    assert_data_eq!(
+        complete!(cmd, "--certain_num [TAB]"),
+        snapbox::str![
+            "val1
+val2
+val3"
+        ]
+    );
+
+    assert_data_eq!(
+        complete!(cmd, "--certain_num val1 [TAB]"),
+        snapbox::str![
+            "--certain_num
+--uncertain_num
+--help\tPrint help
+-Y
+-N
+-h\tPrint help"
+        ]
+    );
+
+    assert_data_eq!(
+        complete!(cmd, "--certain_num [TAB]"),
+        snapbox::str![
```

Comment:
Let's also test "one past the end" to make sure our counting is right

Reply:
Okay.